vyos/vyos-1x
# dhcp-server: T4344: Fix underscores for shared network name

## Change Summary

Shared network name should not be handled by tag node mangling
I.e. should not replace underscores with dashed
## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T4344

## Component(s) name

dhcp, dhcpv6
## Proposed changes


## How to test
Add dhcp_pool with underscores
```
set service dhcp-server shared-network-name NET_01 authoritative
set service dhcp-server shared-network-name NET_01 name-server '1.1.1.1'
set service dhcp-server shared-network-name NET_01 subnet 192.0.2.0/24 range R1 start '192.0.2.21'
set service dhcp-server shared-network-name NET_01 subnet 192.0.2.0/24 range R1 stop '192.0.2.254'
```

Before fix
dhcp.conf
```
...
    on commit {
        set shared-networkname = "NET-01";
    }
```
After fix
Expected the same name in dhcp configuration:
```
# Shared network configration(s)
shared-network NET_01 {
    authoritative;
    option domain-name-servers 1.1.1.1;
    subnet 192.0.2.0 netmask 255.255.255.0 {
        default-lease-time 86400;
        max-lease-time 86400;
        pool {
            range 192.0.2.21 192.0.2.254;
        }
    }
    on commit {
        set shared-networkname = "NET_01";
    }
}
```

## Checklist:
